app/services/concerns/course/user_invitation_service/parse_invitation_concern.rb

```diff
@@ -29,7 +33,9 @@ def parse_invitations(users)
       else
         parse_from_form(users)
       end
-    result.each { |user| user[:email] = user[:email].downcase }
+    users = result.each { |user| user[:email] = user[:email].downcase }
+    emails = users.map { |user| user[:email] }
+    users.partition { |user| emails.count(user[:email]) == 1 }
```

Comment:
This is an O(n^2) operation which could be done in O(n) using a simple hashset (granted, the csv is probably small).